### ISSUE TITLE
Downloader should not report checking of existing files as "downloading"

### ISSUE THREAD

**Issue by wmitsuda** [erigontech/erigon#13250]
This is a low priority, but very quality-of-life improvement request:

When for some reason (bug, republishing, etc...) you need to manually remove some snapshot files, during the check process the downloader does against existing files, they are reported as "downloading" with a very high download speed, which is obviously not true :)

Ideally, the download should be split into some checking/download process (I'm not familiar with that code, so not sure if it is already that way) and the logs should be more exactly regarding what it is doing because now it is confusing.

Example logs from the restore I'm running right now, it is probably checking about 1.7TB of existing files before starting the download of the rest:

```
INFO[12-27|00:53:29.132] [snapshots] no progress yet              files=87 list=accessor/v1-logtopics.0-64.efi,accessor/v1-storage.640-704.efi
,accessor/v1-storage.832-896.efi,idx/v1-code.1664-1696.ef,accessor/v1-code.256-320.vi,...
INFO[12-27|00:53:48.392] [1/6 OtterSync] Downloading              progress="(1111/1111 files) 42.09% - 733.3GB/1.7TB" rate=3.1GB/s time-left=0
hrs:5m total-time=4m20s download-rate=3.1GB/s completion-rate=3.2GB/s alloc=3.2GB sys=6.7GB
INFO[12-27|00:53:51.048] [snapshots] no progress yet              files=87 list=accessor/v1-logtopics.0-64.efi,accessor/v1-storage.640-704.efi
,accessor/v1-storage.832-896.efi,accessor/v1-storage.0-64.vi,idx/v1-code.1664-1696.ef,...
INFO[12-27|00:53:51.827] P2P                                      app=caplin peers=8
INFO[12-27|00:54:08.385] [1/6 OtterSync] Downloading              progress="(1111/1111 files) 45.70% - 796.2GB/1.7TB" rate=3.1GB/s time-left=0
hrs:5m total-time=4m40s download-rate=3.1GB/s completion-rate=3.2GB/s alloc=5.5GB sys=6.9GB
INFO[12-27|00:54:09.945] [snapshots] no progress yet              files=87 list=accessor/v1-storage.448-512.efi,idx/v1-code.1344-1408.ef,idx/v
1-code.1088-1152.ef,idx/v1-accounts.704-768.ef,accessor/v1-tracesfrom.0-64.efi,...
```
